tests/unit/components/search/SearchComponent.spec.ts

```diff
@@ -41,7 +41,7 @@ describe('Search Component', () => {
         setTimeout(() => {
           expect(wrapper.emitted().searchChanged[0][0]).toEqual('dem')
           done()
-        }, 300)
+        }, 600)
```

Comment:
Most timeouts can be removed when using leading: true on the debounce:

```
diff --git a/src/components/search/SearchComponent.vue b/src/components/search/SearchComponent.vue
index 75f544a..fc64294 100644
--- a/src/components/search/SearchComponent.vue
+++ b/src/components/search/SearchComponent.vue
@@ -50,7 +50,7 @@ export default Vue.extend({
           this.lastSearched = ''
         }
       }
-    }, 300)
+    }, 300, { leading: true })
   }
 })
 </script>
diff --git a/tests/unit/components/search/SearchComponent.spec.ts b/tests/unit/components/search/SearchComponent.spec.ts
index 75304df..5f3454b 100644
--- a/tests/unit/components/search/SearchComponent.spec.ts
+++ b/tests/unit/components/search/SearchComponent.spec.ts
@@ -38,10 +38,8 @@ describe('Search Component', () => {
       beforeEach(() => wrapper.find('input').setValue('dem'))
 
       it('should emit a search val changed event with value dem', (done) => {
-        setTimeout(() => {
-          expect(wrapper.emitted().searchChanged[0][0]).toEqual('dem')
-          done()
-        }, 600)
+        expect(wrapper.emitted().searchChanged[0][0]).toEqual('dem')
+        done()
       })
     })
 
@@ -66,10 +64,8 @@ describe('Search Component', () => {
       })
 
       it('should emit a empty search invent, clearing the search', (done) => {
-        setTimeout(() => {
-          expect(wrapper.emitted()).toEqual({})
-          done()
-        }, 600)
+        expect(wrapper.emitted()).toEqual({})
+        done()
       })
     })
   })
(END)

```

Reply:
that is not the desired behaviour, we want give allow the user to finish typing